### DISCUSSION TITLE
Link home-made php script to addtorrent.php

### DISCUSSION THREAD

**Discussion by Hightmar** [Novik/ruTorrent#2319] [Q&A]
Hello, I am making a script to add a torrent from a self-made form.
I have to do it like this, Rutorrent and Rtorrent are on a machine that uses a VPN. So I'm going through another machine which is on the same local network.

In my script I want call addtorrent.php so that the torrent is added automatically to Rtorrent and is started. I'm not very good in code and I don't understand what I have to send as information to the $_REQUEST variable (I think it is this one). I tried to give it the $_FILE information but it doesn't seem to be that.

I manage to send my file to the server but when the script calls addtorrent.php I get "File not found."
Result URL :
[https://domain.tld/upload/addtorrent.php?result[]=Failed&name[]=file.name.torrent&](https://domain.tld/upload/addtorrent.php?result%5B%5D=Failed&name%5B%5D=file.name.torrent&)

Anyone know how I can link my script correctly to addtorrent.php please?

[My script](https://github.com/Hightmar/upload-torrent/blob/main/upload.php)

**Comment by stickz** [comment]
You can use IPTables to host ruTorrent outside of the VPN. This way you don't need a script to perform any ruTorrent tasks. If you want to hide the IP address of the machine for ruTorrent, it's possible to host it on cloudflare for free.

**Reply by Hightmar** [reply]
I install RuTorrent and RTorrent on the same machine and with IPTables , I allow access to RuTorrent via Internet without going through the VPN?

With my VPN I can exclude the local network from VPN traffic but the idea is to be able to access it when I'm not at home, hence the script. 
I don't see how to do that with IPTables.

edit : For Cloudflare I'm not sure I understand, are you telling me to use their DNS servers to mask the IP address? 
If that's the case it's true that I didn't think of this solution. I use it for my sites as well. 

I still prefer the VPS solution to limit the risks when using torrents...

**Reply by stickz** [reply]
You can allow access to ruTorrent via the internet without going through the VPN. Then route rTorrent traffic which distributes the torrents through a VPN to limit your risks. A VPS provider can cancel your service. You're better off to use a third-party VPN provider.

I have a working configuration on Private Internet Access with wireguard for this.
https://github.com/pia-foss/manual-connections/issues/62#issuecomment-748414682

**Reply by Hightmar** [reply]
Thanks for your answer, I'll look into it and try to set it up, but I can't do it right away, I'll do it on my new Raspberry Pi that I'm waiting for. 

In the meantime, do you know how I can use `addtorrent.php` or not? I think it shouldn't be too complicated. I don't understand why the script doesn't take the data that are in `$_FILES['torrent_file']`, I used the same naming to make things easier.

When I call `addtorrent.php` from my form directly it works.
-> `array(2) { ["result"]=> array(1) { [0]=> string(7) "Success" } ["name"]=> array(1) { [0]=> string(53) myFile.torrent" } }`

The `$_REQUEST` and `$_FILES` variables are superglobals, so I don't understand why they don't pass from my script to `addtorrent.php`

**Comment by Hightmar** [comment]
I really don't understand.
I do a `var_dump` of `$_REQUEST` and `$_FILES` on `addtorrent.php` with a `die;` right after to stop the script. 

Here is the result when I use **my script** to call `addtorrent.php`

var_dump Request
`array(1) { ["submit"]=> string(0) "" }`
var_dump Files
`array(1) { ["torrent_file"]=> array(5) { ["name"]=> array(1) { [0]=> string(86) "myfile.torrent" } ["type"]=> array(1) { [0]=> string(24) "application/x-bittorrent" } ["tmp_name"]=> array(1) { [0]=> string(14) "/tmp/phpZv2i4o" } ["error"]=> array(1) { [0]=> int(0) } ["size"]=> array(1) { [0]=> int(203516) } } }`

and the result when I call `addtorrent.php` **directly**.

var_dump Request
`array(1) { ["submit"]=> string(0) "" }`
var_dump Files
`array(1) { ["torrent_file"]=> array(5) { ["name"]=> array(1) { [0]=> string(86) "myfile.torrent" } ["type"]=> array(1) { [0]=> string(24) "application/x-bittorrent" } ["tmp_name"]=> array(1) { [0]=> string(14) "/tmp/phpviKVkV" } ["error"]=> array(1) { [0]=> int(0) } ["size"]=> array(1) { [0]=> int(203516) } } } 
`
The data sent are exactly the same. Except that **when I use my script**, the result is `File not found.`
